[Daniel Iwan Kurniawan] iP

Daniel-Kur opened the pull request:
DukePro
“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
FAST SUPER FAST to use
All you need to do is,

download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
double-click it.
add your tasks.
let it manage your tasks for you 😉
And it is FREE!

Features:
- [x] Managing tasks
- [x] Keyword search
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Daniel-Kur committed:
no message

Daniel-Kur committed:
no message

Daniel-Kur committed:
no message

Daniel-Kur committed:
no message

Daniel-Kur committed:
no message

Daniel-Kur committed:
no message

Daniel-Kur committed:
no message

Daniel-Kur committed:
no message

Daniel-Kur committed:
no message

Daniel-Kur committed:
no message

anunayajoshi commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadlines extends Task {
```
Do you think the Deadlines Class should be named "Deadline" or "Deadlines"?

anunayajoshi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,120 @@
+                    System.out.println("\t" + deadlines.toString());
+                    System.out.println("\t" + "Now you have " + list.size() + " tasks in the list");
+                    System.out.println(LINE + "\n");
+                }catch(Exception e) {
```
Perhaps you could add spacing between the curly bracket and the catch word, to better follow coding standards.

anunayajoshi commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,15 @@
+public class Events extends Task {
```
Similar comment to above, should it be Events or Event.

anunayajoshi commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,29 @@
+    public String toString(){
+        if (isMarked){
+            return MARKED + " " + taskName;
+        }else{
```
There should be a whitespace between the curly braces and your words

anunayajoshi commented on src/main/java/ToDos.java:
```diff
@@ -0,0 +1,10 @@
+public class ToDos extends Task {
+    public ToDos(String toDosName){
```
the name of the "toDosName" might be a bit confusing. Do you think there is a better name for it?

anunayajoshi commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,120 @@
+                    System.out.println("\t" + todos.toString());
+                    System.out.println("\t" + "Now you have " + list.size() + " tasks in the list");
+                    System.out.println(LINE + "\n");
+                }catch(Exception e){
```
You can use the checkstyle plugin to fix the whitespace issues.

anunayajoshi commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,5 @@
+public class DukeException extends Exception {
+    public DukeException(String err){
```
Do add a whitespace between the closing bracket and the opening curly bracket

anunayajoshi reviewed (commented):
Some recurring issues about whitespace but code quality is good with a few questionable naming words used.

Daniel-Kur committed:
Added save functionality

Daniel-Kur committed:
Added 'save' functionality

Daniel-Kur committed:
Merge branch 'branch-Level-7'

Daniel-Kur committed:
Added dates feature

Daniel-Kur committed:
Merge branch 'branch-level-8'

Daniel-Kur committed:
Implement OOP

Daniel-Kur committed:
Merge branch 'branch-A-MoreOOP'

Daniel-Kur committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Daniel-Kur committed:
add JUnit tests for TaskList

Daniel-Kur committed:
Change shadowJar version to 7.0.0

Daniel-Kur committed:
add javadoc to public classes and methods

Daniel-Kur committed:
Merge branch 'branch-A-JavaDoc'

Daniel-Kur committed:
Fix coding standard errors

Daniel-Kur committed:
Fix typo in javadoc

Daniel-Kur committed:
Merge branch 'branch-A-CodingStandard'

Daniel-Kur committed:
Change the shadowJar version back to 5.1.0

Daniel-Kur committed:
Add find command

Daniel-Kur committed:
Merge branch 'branch-level-9'

Daniel-Kur committed:
Add javadoc for find function

Daniel-Kur committed:
Fix style error

Daniel-Kur committed:
Merge branch 'branch-A-CheckStyle'

Daniel-Kur committed:
Add GUI

Daniel-Kur committed:
Merge branch 'branch-level-10'

Daniel-Kur committed:
Add assertions

Daniel-Kur committed:
Commands implementation is not maximal

Lack of efficiency, cannot modify the current tasklist and have to
undergo Delete and Add command to modify

Add Update implementation to overcome this issue

Let's further update the implementation to become more efficient

Daniel-Kur committed:
Merge branch 'branch-BCD-Extension'

Daniel-Kur committed:
Add user guide and Ui picture

Daniel-Kur committed:
Fix checkstyle

Daniel-Kur committed:
Fix gradle UI

Daniel-Kur committed:
Update UI

Daniel-Kur committed:
Fix invalid input handling

Daniel-Kur committed:
Fix checkstyle

Daniel-Kur committed:
Fix exception handling
